### PR TITLE
SDK-4713: Extract utils

### PR DESCRIPTION
**Pull request by VladislavStrelchenko**
Ticket: https://spryker.atlassian.net/browse/SDK-4713

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# tooling
+vendor/
+!integrator/project_original/vendor/
+!general/project_original/vendor/
+!generator/project_original/vendor/
+composer.lock
+!integrator/project_original/composer.lock
+!general/project_original/composer.lock
+!generator/project_original/composer.lock
```
Comment (DmytroKlymanSpryker) [inline .gitignore]: Following lines can be removed:

!integrator/project_original/vendor/
!general/project_original/vendor/
!generator/project_original/vendor/
!integrator/project_original/composer.lock
!general/project_original/composer.lock
!generator/project_original/composer.lock

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "symfony/filesystem": "^5.3",
+    "symfony/finder": "^5.3",
+    "symfony/monolog-bundle": "^3.8",
+    "symfony/process": "^5.0"
+  },
+  "require-dev": {
+    "ext-zip": "*",
+    "phpstan/phpstan": "^1.0.0",
+    "phpunit/phpunit": "^9.5.0",
+    "spryker-sdk/manifest-test-data-provider": "dev-master",
```
Comment (DmytroKlymanSpryker) [inline composer.json]: "spryker-sdk/manifest-test-data-provider": "dev-master",
    
    we don't need it here

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "spryker-sdk/utils",
+  "description": "Spryker SDK Utils Module",
+  "type": "library",
+  "license": "proprietary",
+  "require": {
+    "php": ">=7.4",
+    "ext-mbstring": "*",
+    "composer/semver": "^3.3",
```
Comment (DmytroKlymanSpryker) [inline composer.json]: I think we also don't need "composer/semver": "^3.3" here. I think it's not used anyway here.

Comment (DmytroKlymanSpryker) [reply composer.json]: You can use https://github.com/composer-unused/composer-unused to find the unused composer packages in the project

File: src/Infrastructure/Helper/StringHelper.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Utils\Infrastructure\Helper;
+
+class StringHelper
```
Comment (DmytroKlymanSpryker) [inline src/Infrastructure/Helper/StringHelper.php]: We have already [TextCaseHelper.php](https://github.com/spryker-sdk/utils/pull/1/files#diff-a241f9b88db980e8f435f354341c3c07d68c830a8e553cdd42257b61cefa56c2) that should do the same. Can we keep only 1 helper for that?

File: src/Infrastructure/Filter/CamelCaseToDash.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+     * @param mixed $string
+     *
+     * @return string
+     */
+    public function filter($string): string
+    {
+        $value = preg_replace('/([a-z])([A-Z])/', '$1' . addcslashes($this->getSeparator(), '$') . '$2', $string);
+
+        return is_array($value) ? (string)array_shift($value) : (string)$value;
+    }
```
Comment (DmytroKlymanSpryker) [inline src/Infrastructure/Filter/CamelCaseToDash.php]: If we need this filter, we can keep it as a wrap for helper, but the logic itself should be in 1 place - in 1 helper.

**Review by DmytroKlymanSpryker — CHANGES_REQUESTED** [review]
As described in previous comments

File: .gitignore (hunk 1 above)
Comment (VladislavStrelchenko) [reply .gitignore]: ++ 
thanks

File: composer.json (hunk 2 above)
Comment (VladislavStrelchenko) [reply composer.json]: ++ Thanks

File: composer.json (hunk 3 above)
Comment (VladislavStrelchenko) [reply composer.json]: we need it here ` return Comparator::greaterThanOrEqualTo(...`

File: src/Infrastructure/Helper/StringHelper.php (hunk 4 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/StringHelper.php]: ++

File: src/Infrastructure/Filter/CamelCaseToDash.php (hunk 5 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Filter/CamelCaseToDash.php]: Moved the filter to `\Helper\Filter`

We are using it here ` if ($separateAbbreviation) {
            $filterChain->attach(new CamelCaseToDash());
        } else {
            $filterChain->attach(new CamelCaseToDashWithoutAbbreviation());
        }`

File: composer.json (hunk 3 above)
Comment (DmytroKlymanSpryker) [reply composer.json]: ok, got it

**Review by DmytroKlymanSpryker — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        strategy:
+            fail-fast: false
+            matrix:
+                php-version: [
+                        '7.4',
+                        '8.2',
+                ]
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml]: use v4, please. v2 contains deprecated logic

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml]: please use the latest version https://github.com/spryker-sdk/upgrader/blob/master/.github/workflows/trigger-remote-pr.yml#L24

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                  php-version: ${{ matrix.php-version }}
+                  extensions: mbstring, intl, pdo_mysql
+
+            - name: Validate composer.json and composer.lock
+              run: composer validate
+
+            - name: Get Composer Cache Directory
+              id: composer-cache
+              run: |
+                  echo "::set-output name=dir::$(composer config cache-files-dir)"
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml]: this logic is deprecated.
Please use the following logic --> https://github.com/spryker-sdk/upgrader/blob/master/.github/workflows/trigger-remote-pr.yml#L66-L69. More info https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+            - name: Composer install
+              run: composer install --prefer-dist --no-interaction --optimize-autoloader
+
+            - name: Check PHP for syntax errors
+              run: find ./src -path src -prune -o -type f -name '*.php' -print0 | xargs -0 -n1 -P4 php -l -n | (! grep -v "No syntax errors detected" )
+
+            - name: PHPStan checks
+              run: composer stan
+
+    cs-check:
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml]: we can move `CodeStyle checks` job into validation and remove this step completely.

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+                path: ${{ steps.composer-cache.outputs.dir }}
+                key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+                restore-keys: ${{ runner.os }}-composer-
+
+            - name: Composer install
+              run: composer install --prefer-dist --no-interaction --optimize-autoloader
+
+            - name: Run tests
+              run: |
+                  if [[ ${{ matrix.php-version }} == '7.4' ]]; then
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml]: Let's have `8.1` as a version for code coverage.

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+            - name: Validate composer.json and composer.lock
+              run: composer validate
+
+            - name: Get Composer Cache Directory
+              id: composer-cache
+              run: |
+                echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - name: Composer cache
+              uses: actions/cache@v2
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml]: please use latest `actions/checkout@v3`

File: .github/workflows/ci.yml [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                php-version: ${{ matrix.php-version }}
+                extensions: mbstring, intl, pdo_mysql
+
+            - name: Validate composer.json and composer.lock
```
Comment (pavelmaksimov25) [inline .github/workflows/ci.yml]: We don't need to duplicate this step. Already present in the `validation` job

File: phpstan.neon [hunk 13]
```diff
@@ -0,0 +1,9 @@
+parameters:
+    level: 8
+    paths:
+        - src/
+    bootstrapFiles:
+        - tests/bootstrap.php
```
Comment (pavelmaksimov25) [inline phpstan.neon]: why in `tests/` ?

File: src/Infrastructure/Helper/TextCaseHelper.php [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+     * @param string $value
+     * @param bool $separateAbbreviation
+     *
+     * @return string
+     */
+    public static function camelCaseToDash(string $value, bool $separateAbbreviation = true): string
+    {
+        $filterChain = new FilterChain();
+
+        if ($separateAbbreviation) {
```
Comment (sergeyspryker) [inline src/Infrastructure/Helper/TextCaseHelper.php]: avoid else
```
$camelCaseToDashFilter = $separateAbbreviation
    ? new CamelCaseToDash()
    : new CamelCaseToDashWithoutAbbreviation();

$filterChain->attach($camelCaseToDashFilter);
```

**Review by vol4onok — APPROVED** [review]

File: src/Infrastructure/Helper/TextCaseHelper.php [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+{
+    /**
+     * @param string $value
+     * @param bool $separateAbbreviation
+     *
+     * @return string
+     */
+    public static function camelCaseToDash(string $value, bool $separateAbbreviation = true): string
+    {
+        $filterChain = new FilterChain();
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Helper/TextCaseHelper.php]: Why do we need to use laminas when we can use a lightweight raw PHP functions?
Example -> https://github.com/illuminate/support/blob/master/Str.php#L1310-L1325

File: src/Infrastructure/Service/FinderFactory.php [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Utils\Infrastructure\Service;
+
+use Symfony\Component\Finder\Finder;
+
+class FinderFactory
```
Comment (pavelmaksimov25) [inline src/Infrastructure/Service/FinderFactory.php]: Looks like an obsolete factory, isn't it?

File: src/Infrastructure/Helper/SemanticVersionHelper.php [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+     * @return int|null
+     */
+    public static function getMajorVersion(string $semanticVersion): ?int
+    {
+        if (!static::isSemanticVersion($semanticVersion)) {
+            return null;
+        }
+
+        $versionParts = explode('.', $semanticVersion);
+        if (!$versionParts) {
```
Comment (sergeyspryker) [inline src/Infrastructure/Helper/SemanticVersionHelper.php]: it's never been an empty array

File: src/Infrastructure/Helper/SemanticVersionHelper.php [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+            return null;
+        }
+
+        return (int)array_shift($versionParts);
```
Comment (sergeyspryker) [inline src/Infrastructure/Helper/SemanticVersionHelper.php:29]: just use `(int)$versionParts[0]`

File: src/Infrastructure/Helper/SemanticVersionHelper.php [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * @param string $version
+     *
+     * @return bool
+     */
+    protected static function isSemanticVersion(string $version): bool
+    {
+        return Comparator::greaterThanOrEqualTo($version, '0.0.1') === true;
```
Comment (sergeyspryker) [inline src/Infrastructure/Helper/SemanticVersionHelper.php]: don't need comparing with true
`return Comparator::greaterThanOrEqualTo($version, '0.0.1')`

File: src/Infrastructure/Helper/TextCaseHelper.php [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+     * @param bool $upperCaseFirst
+     *
+     * @return string
+     */
+    public static function dashToCamelCase(string $value, bool $upperCaseFirst = true): string
+    {
+        $filterChain = new FilterChain();
+        $filterChain->attach(new DashToCamelCase());
+
+        if ($upperCaseFirst) {
```
Comment (sergeyspryker) [inline src/Infrastructure/Helper/TextCaseHelper.php]: Minor: don't need double filter call
`$filterResult = $filterChain->filter($value)`

File: src/Infrastructure/Helper/TextCaseHelper.php [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+        ]);
+    }
+
+    /**
+     * @param string $string
+     * @param bool $capitalizeFirstCharacter
+     *
+     * @return string
+     */
+    public static function fromDashToCamelCase(string $string, bool $capitalizeFirstCharacter = true): string
```
Comment (sergeyspryker) [inline src/Infrastructure/Helper/TextCaseHelper.php]: What's the different with `dashToCamelCase` ?

File: src/Infrastructure/Service/ProcessRunnerService.php [hunk 22]
```diff
@@ -0,0 +1,115 @@
...
+{
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected LoggerInterface $logger;
+
+    /**
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
```
Comment (sergeyspryker) [inline src/Infrastructure/Service/ProcessRunnerService.php:25]: Logger should be optional (nullable). Let assume we want to use it aside of symfony
create method debug with check if logger set or not. Also composer should require psr/logger package instead of monolog

File: src/Infrastructure/Service/ProcessRunnerService.php [hunk 23]
```diff
@@ -0,0 +1,115 @@
...
+    public function runFromCommandLine(
+        string $command,
+        ?string $cwd = null,
+        ?array $env = null,
+        $input = null,
+        ?float $timeout = self::DEFAULT_PROCESS_TIMEOUT
+    ): Process {
+        $this->logger->debug(sprintf('Run command: %s', $command), ['cwd' => $cwd, 'env' => $env]);
+        $process = Process::fromShellCommandline($command, $cwd, $env, $input, $timeout);
+        $process->setTimeout(static::DEFAULT_PROCESS_TIMEOUT);
```
Comment (sergeyspryker) [inline src/Infrastructure/Service/ProcessRunnerService.php]: Timeout is already set above

File: src/Infrastructure/Service/ProcessRunnerService.php [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+     * @param array<string> $command
+     * @param array<string, mixed> $env
+     *
+     * @return \Symfony\Component\Process\Process<string, string>
+     */
+    public function run(array $command, array $env = []): Process
+    {
+        $this->logger->debug(sprintf('Run command: %s', implode(' ', $command)), $env);
+        $process = new Process($command, (string)getcwd(), $env);
+        $process->setTimeout(static::DEFAULT_PROCESS_TIMEOUT);
```
Comment (sergeyspryker) [inline src/Infrastructure/Service/ProcessRunnerService.php]: Set timeout in Process construct

File: src/Infrastructure/Service/ProcessRunnerService.php [hunk 25]
```diff
@@ -0,0 +1,115 @@
...
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param array<string> $command
+     * @param array<string, mixed> $env
+     *
+     * @return \Symfony\Component\Process\Process<string, string>
+     */
+    public function run(array $command, array $env = []): Process
```
Comment (sergeyspryker) [inline src/Infrastructure/Service/ProcessRunnerService.php]: Use the full signature of `Process` construct

File: tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php [hunk 26]
```diff
@@ -0,0 +1,101 @@
...
+            ['test-foo-bar', 'TestFooBar', true],
+            ['test-foo-bar', 'TestFooBar', false],
+            ['test-foo-bar', 'test-Foo-Bar', true],
+            ['test-foo-bar', 'test-Foo-Bar', false],
+            ['test-foo-bar-abc-test', 'testFooBarABCTest', true],
+            ['test-foo-bar-abctest', 'testFooBarABCTest', false],
+        ];
+
+        foreach ($data as $set) {
+            yield $set;
```
Comment (sergeyspryker) [inline tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php]: There is no profit from the generator usage coz the `$data` variable is used for storing the whole array im memory.
Usually is used like that
```
yield ['test-foo-bar', 'testFooBar', true];
yield ['test-foo-bar', 'testFooBar', false];
yield ['test-foo-bar', 'TestFooBar', true];
...
```
From my perspective generators are overhead for tests dataproviders

File: tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+            ['testFooBar', 'test-foo-bar', false],
+            ['TestFooBar', 'Test-foo-bar', false],
+            ['TestFooBarAbcTest', 'test-foo-bar-abc-test', true],
+            ['testFooBarAbctest', 'test-foo-bar-abctest', false],
+            ['TestFooBarAbc123Test', 'test-foo-bar-abc123-test', true],
+            ['TestFooBarAbc123Test', 'test-foo-bar-abc-123-test', true],
+            ['TestFooBarAbc123test', 'test-foo-bar-abc-123test', true],
+        ];
+
+        foreach ($data as $set) {
```
Comment (sergeyspryker) [inline tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php]: The same

File: tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+     *
+     * @param string $expResult
+     * @param string $value
+     * @param bool $upperCaseFirst
+     *
+     * @return void
+     */
+    public function testDashToCamelCase(string $expResult, string $value, bool $upperCaseFirst): void
+    {
+        $this->assertSame($expResult, TextCaseHelper::dashToCamelCase($value, $upperCaseFirst));
```
Comment (sergeyspryker) [inline tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php]: Minor: make this test with result variable as below with `Act` and `Assert` sections

File: tests/bootstrap.php [hunk 29]
```diff
@@ -0,0 +1,3 @@
+<?php
+
+define('APPLICATION_ROOT_DIR', dirname(__DIR__, 1));
```
Comment (sergeyspryker) [inline tests/bootstrap.php]: What this constant is used for?

File: .github/workflows/ci.yml (hunk 6 above)
Comment (VladislavStrelchenko) [reply .github/workflows/ci.yml]: ++

File: .github/workflows/ci.yml (hunk 7 above)
Comment (VladislavStrelchenko) [reply .github/workflows/ci.yml]: ++

File: phpstan.neon [hunk 30]
```diff
@@ -0,0 +1,9 @@
...
+    checkMissingIterableValueType: false
+    checkGenericClassInNonGenericObjectType: false
+    treatPhpDocTypesAsCertain: false
```
Comment (sergeyspryker) [inline phpstan.neon]: Why did you add these rules exceptions?

**Review by sergeyspryker — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 10 above)
Comment (VladislavStrelchenko) [reply .github/workflows/ci.yml]: ++

File: .github/workflows/ci.yml (hunk 11 above)
Comment (VladislavStrelchenko) [reply .github/workflows/ci.yml]: ++

File: .github/workflows/ci.yml (hunk 12 above)
Comment (VladislavStrelchenko) [reply .github/workflows/ci.yml]: removed

File: phpstan.neon (hunk 13 above)
Comment (VladislavStrelchenko) [reply phpstan.neon]: removed
we need it only for test

File: src/Infrastructure/Service/FinderFactory.php (hunk 16 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Service/FinderFactory.php]: Removed it

File: src/Infrastructure/Helper/TextCaseHelper.php (hunk 15 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/TextCaseHelper.php]: I have extracted logic from the upgrader and do not expect to totally rewrite it

File: src/Infrastructure/Helper/TextCaseHelper.php (hunk 14 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/TextCaseHelper.php]: ++

File: .github/workflows/ci.yml (hunk 8 above)
Comment (VladislavStrelchenko) [reply .github/workflows/ci.yml]: ++

File: .github/workflows/ci.yml (hunk 9 above)
Comment (VladislavStrelchenko) [reply .github/workflows/ci.yml]: ++

File: src/Infrastructure/Helper/TextCaseHelper.php (hunk 15 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/TextCaseHelper.php]: Discussed

File: src/Infrastructure/Helper/SemanticVersionHelper.php (hunk 18 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/SemanticVersionHelper.php:29]: I do not like undescribed keys

File: src/Infrastructure/Helper/SemanticVersionHelper.php (hunk 17 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/SemanticVersionHelper.php]: I prefer this more safety approach

File: src/Infrastructure/Helper/SemanticVersionHelper.php (hunk 19 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/SemanticVersionHelper.php]: ++

File: src/Infrastructure/Helper/TextCaseHelper.php (hunk 20 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/TextCaseHelper.php]: ++

File: src/Infrastructure/Helper/TextCaseHelper.php (hunk 21 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/TextCaseHelper.php]: Remove
Thanks

File: src/Infrastructure/Helper/SemanticVersionHelper.php (hunk 17 above)
Comment (sergeyspryker) [reply src/Infrastructure/Helper/SemanticVersionHelper.php]: Explain how to cover that by phpunit test

File: src/Infrastructure/Service/ProcessRunnerService.php (hunk 23 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Service/ProcessRunnerService.php]: ++

File: src/Infrastructure/Service/ProcessRunnerService.php (hunk 24 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Service/ProcessRunnerService.php]: ++

File: src/Infrastructure/Service/ProcessRunnerService.php (hunk 25 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Service/ProcessRunnerService.php]: ++

File: tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php (hunk 26 above)
Comment (VladislavStrelchenko) [reply tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php]: From my perspective, generators provide clear structure

File: tests/bootstrap.php (hunk 29 above)
Comment (VladislavStrelchenko) [reply tests/bootstrap.php]: deleted

File: phpstan.neon (hunk 30 above)
Comment (VladislavStrelchenko) [reply phpstan.neon]: Our default rules

File: src/Infrastructure/Helper/SemanticVersionHelper.php (hunk 17 above)
Comment (VladislavStrelchenko) [reply src/Infrastructure/Helper/SemanticVersionHelper.php]: ++

File: tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php (hunk 26 above)
Comment (VladislavStrelchenko) [reply tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php]: ++

File: tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php (hunk 27 above)
Comment (VladislavStrelchenko) [reply tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php]: ++

File: tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php (hunk 28 above)
Comment (VladislavStrelchenko) [reply tests/Utils/Infrastructure/Helper/TextCaseHelperTest.php]: discussed

**Review by sergeyspryker — APPROVED** [review]

**Review by DmytroKlymanSpryker — APPROVED** [review]
